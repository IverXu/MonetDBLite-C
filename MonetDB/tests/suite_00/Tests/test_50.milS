setoid(oid(20000000));
# Selfcontained test of delta operations
module("trans");
b:=new(int,int);
bi:=new(int,int);
bd:=new(int,int);
insert(b,1,2);
insert(b,2,3);
bd:=b.delta;
print(bd);
a:= b.alpha;
a.print;

#commit the changes and repeat the action
clear(bd);
commit(b);
b.alpha.print;

insert(b,4,5);
delete(b,2,3);
print(b);
bd:=b.delta();
print(bd);
a:= b.alpha;
a.print;
abort(b);
print(b);
quit;
