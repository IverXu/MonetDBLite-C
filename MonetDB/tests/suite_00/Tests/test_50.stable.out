stdout of test 'test_50` in directory 'tests/suite_00` itself:

#> Mtest:	milS test_50 milS
#> Mtest:	Mtimeout -timeout 210 Mserver -modulesilent -yaccsilent -d8 -db tests_suite_00 -single test_50.milS
# Monet Database Server V4.1
# Copyright (c) 1993-1999, CWI & SION. All rights reserved.
# compiled for IRIX646.5/32bit on Fri Jan 14 19:36:03 MET 2000.
#script:test_50.milS

loaded;
#---------------------------------#
# module	| usage_count	  #
#---------------------------------#
[ "algebra",	  1		  ]
[ "arith",	  1		  ]
[ "bat",	  1		  ]
[ "kernel",	  1		  ]
[ "str",	  1		  ]
[ "sys",	  1		  ]
[ "trans",	  1		  ]

Try: modules;                - for other modules.
     ls;                     - for persistent bat listing.
     sigs("<module>")        - for available commands per module.
     help("<command>")       - for help one-liner for a command.

#You are system administrator (adm)!
setoid(oid(20000000));
# Selfcontained test of delta operations
module("trans");
b:=new(int,int);
bi:=new(int,int);
bd:=new(int,int);
insert(b,1,2);
insert(b,2,3);
bd:=b.delta;
print(bd);
#-----------------#
# BAT:	  tmp_28  #
# (int)	  (int)	  #
#-----------------#
a:= b.alpha;
a.print;
#-----------------#
# BAT:	  tmp_27  #
# (int)	  (int)	  #
#-----------------#
[ 1,	  2	  ]
[ 2,	  3	  ]

#commit the changes and repeat the action
clear(bd);
commit(b);
b.alpha.print;
#-----------------#
# BAT:	  tmp_29  #
# (int)	  (int)	  #
#-----------------#

insert(b,4,5);
delete(b,2,3);
print(b);
#-----------------#
# BAT:	  tmp_25  #
# (int)	  (int)	  #
#-----------------#
[ 1,	  2	  ]
[ 4,	  5	  ]
bd:=b.delta();
print(bd);
#-----------------#
# BAT:	  tmp_29  #
# (int)	  (int)	  #
#-----------------#
[ 2,	  3	  ]
a:= b.alpha;
a.print;
#-----------------#
# BAT:	  tmp_28  #
# (int)	  (int)	  #
#-----------------#
[ 4,	  5	  ]
abort(b);
print(b);
#-----------------#
# BAT:	  tmp_25  #
# (int)	  (int)	  #
#-----------------#
[ 2,	  3	  ]
[ 1,	  2	  ]
quit;
